hoomd/md/TwoStepBD.cc

```diff
@@ -84,7 +84,8 @@ void TwoStepBD::integrateStepOne(unsigned int timestep)
         unsigned int ptag = h_tag.data[j];
 
         // Initialize the RNG
-        RandomGenerator rng(RNGIdentifier::TwoStepBD, m_seed, ptag, timestep);
+        RandomGenerator rng(hoomd::Seed(RNGIdentifier::TwoStepBD, timestep, seed),
```

Comment:
I had not thought about that side effect of centralizing the seed. I will add some documentation to the `Seed` class regarding this.

Reply:
Classes where this could apply:
* HPMC integrators - These are effectively singletons as there is only one integrator per Simulation. I don't think this is an issue.
* HPMC BoxMC - This should be used as a singleton, but in reality users *could* instantiate multiple BoxMC updaters.
* Same for HPMC Cluster, MuVT, and QuickCompress updaters.
* mpcd ColissionMethod gridshift

@glotzerlab/hoomd-api-reviewers what do you think on the HPMC singleton updaters. Should we force them to be singletons in the Python API? Or should we allow multiple and communicate their position in the Updaters list to use as an instance id? Do you have another suggestion?

The thing I don't like about using a static variable solution is that then the seed is dependent on the number of classes and the order in which they are initialized. It is also global and not tied to the specific Simulation. If we want to go that route, I might just use the `this` pointer as an instance id - however that will not be reproducible.

Reply:
@b-butler and I discussed this offline. We will add back in a settable instance id for these classes so that users can set unique values in the rare case that they are required.